src/nvim/spell.c

```diff
@@ -1283,6 +1299,8 @@ size_t spell_move_to(win_T *wp, int dir, bool allwords, bool curline, hlf_T *att
     }
     char_u *p = buf + skip;
     char_u *endp = buf + len;
+    decor_redraw_line(wp->w_buffer, lnum - 1, &decor_state);
+    decor_providers_invoke_spell(wp, lnum - 1, col, lnum - 1, -1, &decor_error);
```

Comment:
Swapping these two makes a little more sense? Even if not needed for now...

Reply:
I've copied the order we have in `drawline.c` for `decor_redraw_line` and `decor_providers_invoke_line`.

But thinking about it, you might be right since `decor_redraw_line` populates `decor_state.active` which should be done after all extmarks have been set.